File: PRB2b/solutions/sol_slow.cpp
#pragma GCC optimize("Ofast")
#pragma GCC target("avx,avx2,fma")

#include <iostream>
#include <algorithm>

using namespace std;

int main() {
    int N, K;
    cin >> N;
    // Store point values in an array
    int* points = new int[N];
    long long sum = 0;
    for (int i = 0; i < N; i++) {
        cin >> points[i];
        sum += points[i];
    }
    cin >> K;
    for (int i = 0; i < K; i++) {
        int minI = 0;
        for (int j = 1; j < N; j++) {
            if (points[j] < points[minI]) {
                minI = j;
            }
        }
        sum -= points[minI];
        // Swap min with last element
        points[minI] = points[--N];
    }
    cout << sum << endl;
    return 0;
}

File: PRB2b/solutions/sol_slow.py
N = int(input())
points = list(map(int, input().split()))
K = int(input())

curr_sum = 0
if K < N / 2:
    curr_sum = sum(points)
    for i in range(K):
        x = min(points)
        curr_sum -= x
        points.remove(x)
else:
    for i in range(N-K):
        x = max(points)
        curr_sum += x
        points.remove(x)

print(curr_sum)
